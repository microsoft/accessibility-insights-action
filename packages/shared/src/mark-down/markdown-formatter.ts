// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { brand, brandLogoImg, toolName } from '../content/strings';

export const escaped = (markdown: string): string => {
    return markdown.replace(/</g, '\\<'); // CodeQL [SM02383] False Positive: The expected input does not include a backslash, so the replacement is not needed.
};

export const snippet = (text: string): string => {
    return `\`${text}\``;
};

export const link = (href: string, text: string): string => {
    return `[${text}](${href})`;
};

export const image = (altText: string, src: string): string => {
    return `![${altText}](${src})`;
};

export const listItem = (text: string): string => {
    return `* ${text}`;
};

export const heading = (text: string, headingLevel: number): string => {
    return `${'#'.repeat(headingLevel)} ${text}`;
};

export const bold = (text: string): string => {
    return `**${text}**`;
};

export const productTitle = (): string => {
    return `${image(`${brand}`, brandLogoImg)} ${toolName}`;
};

export const footerSeparator = (): string => `---`;

export const sectionSeparator = (): string => '\n';
