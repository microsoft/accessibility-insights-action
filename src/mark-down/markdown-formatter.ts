// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { brand, brandLogoImg, toolName } from '../content/strings';

export const snippet = (text: string): string => {
    return `\`${text}\``;
};

export const link = (href: string, text: string): string => {
    return `[${text}](${href})`;
};

export const image = (altText: string, src: string): string => {
    return `![${altText}](${src})`;
};

export const listItem = (text: string): string => {
    return `* ${text}`;
};

export const heading = (text: string, headingLevel: number): string => {
    return `${'#'.repeat(headingLevel)} ${text}`;
};

export const bold = (text: string): string => {
    return `**${text}**`;
};

export const productTitle = (): string => {
    return `${image(`${brand}`, brandLogoImg)} ${toolName}`;
};

export const footerSeparator = (): string => `---`;

export const sectionSeparator = (): string => '\n';
