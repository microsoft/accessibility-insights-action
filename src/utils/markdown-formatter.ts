// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { brand, brandLogoImg } from '../content/strings';

export const snippet = (text: string): string => {
    return `\`${text}\``;
};

export const link = (href: string, text: string): string => {
    return `[${text}](${href})`;
};

export const image = (altText: string, src: string): string => {
    return `![${altText}](${src})`;
};

export const listItem = (text: string): string => {
    return `* ${text}`;
};

export const heading = (text: string, headingLevel: number) => {
    return `${'#'.repeat(headingLevel)} ${text}`;
};

export const bold = (text: string) => {
    return `**${text}**`;
};

export const productTitle = () => {
    return `${image(`${brand}`, brandLogoImg)} ${brand}`;
};

export const footerSeparator = () => `---`;

export const sectionSeparator = () => '\n';
