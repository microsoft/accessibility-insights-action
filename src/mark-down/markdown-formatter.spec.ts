// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { bold, footerSeparator, heading, image, link, listItem, productTitle, sectionSeparator, snippet } from './markdown-formatter';

describe('MarkdownFormatter', () => {
    it('snippet', () => {
        expect(snippet('code')).toMatchSnapshot();
    });

    it('link', () => {
        expect(link('href', 'text')).toMatchSnapshot();
    });

    it('image', () => {
        expect(image('alt', 'src')).toMatchSnapshot();
    });

    it('listItem', () => {
        expect(listItem('li')).toMatchSnapshot();
    });

    it('heading', () => {
        expect(heading('heading', 2)).toMatchSnapshot();
    });

    it('bold', () => {
        expect(bold('text')).toMatchSnapshot();
    });

    it('productTitle', () => {
        expect(productTitle()).toMatchSnapshot();
    });

    it('footerSeparator', () => {
        expect(footerSeparator()).toMatchSnapshot();
    });

    it('sectionSeparator', () => {
        expect(sectionSeparator()).toMatchSnapshot();
    });
});
