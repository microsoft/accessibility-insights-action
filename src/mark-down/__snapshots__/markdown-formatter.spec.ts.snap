// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownFormatter bold 1`] = `"**text**"`;

exports[`MarkdownFormatter footerSeparator 1`] = `"---"`;

exports[`MarkdownFormatter heading 1`] = `"## heading"`;

exports[`MarkdownFormatter image 1`] = `"![alt](src)"`;

exports[`MarkdownFormatter link 1`] = `"[text](href)"`;

exports[`MarkdownFormatter listItem 1`] = `"* li"`;

exports[`MarkdownFormatter productTitle 1`] = `"![Accessibility Insights](https://accessibilityinsights.io/img/a11yinsights-blue.svg) Accessibility Insights Action"`;

exports[`MarkdownFormatter sectionSeparator 1`] = `
"
"
`;

exports[`MarkdownFormatter snippet 1`] = `"\`code\`"`;
